coil/src/main/java/com/google/accompanist/coil/Coil.kt

```diff
@@ -62,50 +61,58 @@ object CoilImageDefaults {
 }
 
 /**
- * Creates a [CoilAsyncImageState] that is remembered across compositions.
+ * Creates a [CoilImageState] that is remembered across compositions.
  *
  * Changes to the provided values for [imageLoader] and [context] will **not** result
  * in the state being recreated or changed in any way if it has already been created.
- * Changes to [data] and [requestBuilder] will result in the [CoilAsyncImageState] being updated.
+ * Changes to [data], [shouldRefetchOnSizeChange] & [requestBuilder] will result in
+ * the [CoilImageState] being updated.
  *
- * @param data the value for [CoilAsyncImageState.data]
- * @param imageLoader the value for [CoilAsyncImageState.imageLoader]
- * @param context the initial value for [CoilAsyncImageState.context]
- * @param requestBuilder the value for [CoilAsyncImageState.requestBuilder]
+ * @param data the value for [CoilImageState.data]
+ * @param imageLoader the value for [CoilImageState.imageLoader]
+ * @param context the initial value for [CoilImageState.context]
+ * @param shouldRefetchOnSizeChange the value for [CoilImageState.shouldRefetchOnSizeChange]
+ * @param requestBuilder the value for [CoilImageState.requestBuilder]
  */
 @Composable
-fun rememberCoilAsyncImageState(
+fun rememberCoilImageState(
     data: Any?,
     imageLoader: ImageLoader = CoilImageDefaults.defaultImageLoader(),
     context: Context = LocalContext.current,
+    shouldRefetchOnSizeChange: ShouldRefetchOnSizeChange = { _, _ -> false },
```

Comment:
is it valid to define a type with a private typealias? maybe worth writing it in full here

Reply:
It compiles fine, but done anyway to be clearer.